.github/workflows/main.yml

```diff
@@ -30,12 +36,96 @@ jobs:
         run: |
           just test
 
+  lint-dockerfile:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: hadolint/hadolint-action@54c9adbab1582c2ef04b2016b760714a4bfde3cf  # v3.1.0
+        with:
+          dockerfile: docker/Dockerfile
+
+  docker-test:
```

Comment:
This is similar to how job-server does things, we just don't have much of a test harness yet.  The goal here is to check we can smoke test the running container, but since we're building it we also take the time to save it so deploy can save a bit of time too.

Reply:
`metrics-test-and-build` perhaps then? The building and uploading of the image seems like a key part of the action.